### DISCUSSION TITLE
Is there a document listing the values for configuration parameters?

### DISCUSSION THREAD

**Discussion by micleeso** [Sunoo/homebridge-camera-ffmpeg#1362] [Q&A]
Is there a document that defines valid values for configuration parameters? 
i.e. source, vcodec, returnAudioTarget

Thank you.

**Comment by Sunoo** [comment]
There is some documentation on the main page of the project. source and returnAudioTarget will be highly dependent on whatever camera you’re using. vcodec is mostly to allow for hardware accelerated encoding, or disabling transcoding in some situations.

**Reply by micleeso** [reply]
Okay, thank you. 
I understand the purposes of each, but I am looking for valid configuration values.
e.g.
**returnAudioTarget** what would be valid for a Foscam FI9821W V2.1, or How do I find that out?
As for **codec** "copy" is self-explanatory. Other valid configuration values that I am aware of are "h264", "h264_omx", "libx264" thinking these are for hardware-based transcoding but for which hardware platform, I do not know. Are there other options.

I am running Hoobs on a Raspberry pi 4 with 4GB.

Thank you in advance.

**Reply by Sunoo** [reply]
For what would be valid, check the configurations submitted by other users, I can’t test every camera on the market: https://sunoo.github.io/homebridge-camera-ffmpeg/configs/

libx264 is software encoding that will work on anything. h264_omx is for Raspberry Pi’s, but in my experience doesn’t do a great job. h264_videotoolbox is for Macs. There can be other valid options, if you’re using a different build of ffmpeg.
